src/icrar/leap-accelerate/MetaData.cc

```diff
@@ -143,22 +143,30 @@ namespace icrar
      * @param metadata 
      * @param direction 
      */
-    void SetDD(MetaData& metadata, const MVDirection& direction)
+    void MetaData::SetDD(const MVDirection& direction)
     {
-        metadata.dlm_ra = direction.get()[0] - metadata.phase_centre_ra_rad;
-        metadata.dlm_dec = direction.get()[1] - metadata.phase_centre_dec_rad;
+        const int I2D = 2;
 
-        metadata.dd(IPosition(0,0)) = cos(metadata.dlm_ra) * cos(metadata.dlm_dec);
-        metadata.dd(IPosition(0,1)) = -sin(metadata.dlm_ra);
-        metadata.dd(IPosition(0,2)) = cos(metadata.dlm_ra) * sin(metadata.dlm_dec);
+        if(!dd.is_initialized())
+        {
+            dd.reset(casacore::Matrix<double>(3,3));
+        }
+
+        auto& dd3d = dd.value();
+        dlm_ra = direction.get()[0] - phase_centre_ra_rad;
+        dlm_dec = direction.get()[1] - phase_centre_dec_rad;
+
+        dd3d(IPosition(I2D,0,0)) = cos(dlm_ra) * cos(dlm_dec);
```

Comment:
`std::sin` / `std::cos` across all invocations, or `using std::sin` / `using std::cos` inside the function body. Won't make a difference now, but could do if the input type changes in the future.